addons/web/static/src/core/utils/sortable_list.js

```diff
@@ -0,0 +1,360 @@
+ * @property {(params: MoveParams) => any} [onDrop] called when the dragging sequence
+ *  ends on a mouseup action AND the dragged element has been moved elsewhere. The
+ *  callback will be given an object with any useful element regarding the new position
+ *  of the dragged element (@see MoveParams).
+ * @property {(params: SortableListHandlerParams) => any} [onDragEnd] called when the
+ * dragging sequence ends, regardless of the reason.
+ */
+
+/**
+ * @typedef MoveParams
+ * @property {HTMLElement} element
+ * @property {HTMLElement | null} group
+ * @property {HTMLElement | null} previous
+ * @property {HTMLElement | null} next
+ * @property {HTMLElement | null} newGroup
+ * @property {HTMLElement | null} parent
+ * @property {HTMLElement} placeholder
+ */
+
+/**
+ * @typedef SortableState
+ * @property {boolean} dragging
+ */
+
+/** @type {(params: SortableParams) => SortableState} */
+export const useSortableList = makeDraggableHook({
+    name: "useSortableList",
+    acceptedParams: {
+        groups: [String, Function],
+        connectGroups: [Boolean, Function],
+        nest: [Boolean],
+        listType: [String],
+        nestInterval: [Number],
+    },
+    defaultParams: {
+        connectGroups: false,
+        currentGroup: null,
+        cursor: "grabbing",
+        edgeScrolling: { speed: 20, threshold: 60 },
+        elements: "li",
+        groupSelector: null,
+        nest: false,
+        listType: "ul",
+        nestInterval: 15,
+    },
+
+    // Set the parameters.
+    onComputeParams({ ctx, params }) {
+        // Group selector
+        ctx.groupSelector = params.groups || null;
+        if (ctx.groupSelector) {
+            ctx.fullSelector = [ctx.groupSelector, ctx.fullSelector].join(" ");
+        }
+        // Connection across groups
+        ctx.connectGroups = params.connectGroups;
+        // Nested elements
+        ctx.nest = params.nest;
+        // List type
+        ctx.listType = params.listType;
+        // Horizontal distance needed to trigger a change in the list hierarchy
+        // (i.e. changing parent when moving horizontally)
+        ctx.nestInterval = params.nestInterval;
+        ctx.isRTL = localization.direction === "rtl";
+    },
+
+    // Set the current group and create the placeholder row that will take the
+    // place of the moving row.
+    onWillStartDrag({ ctx, addCleanup }) {
+        if (ctx.groupSelector) {
+            ctx.currentGroup = ctx.current.element.closest(ctx.groupSelector);
+            if (!ctx.connectGroups) {
+                ctx.current.container = ctx.currentGroup;
+            }
+        }
+
+        if (ctx.nest) {
+            ctx.prevNestX = ctx.pointer.x;
+        }
+        ctx.current.placeHolder = ctx.current.element.cloneNode(false);
+        ctx.current.placeHolder.style = `display: block; width: 100%; height: 5px; background-color: deepskyblue; padding-top: 0 !important; padding-bottom: 0 !important; background-clip: content-box;`;
```

Comment:
Since you are using backquotes could you split this on 2-3 lines to make it more readable ?